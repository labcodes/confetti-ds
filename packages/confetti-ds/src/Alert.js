import React from "react";
import PropTypes from "prop-types";
import { isUndefined } from "lodash";
import Icon from "./Icon";
import TextButton from "./Button/TextButton";

export default class Alert extends React.Component {
  static propTypes = {
    /** This is the message text string. */
    text: PropTypes.string.isRequired,
    /** Type of the Alert. */
    type: PropTypes.oneOf(["info", "warn", "error"]),
    /** Sets the icon related to the alertâ€™s message. */
    icon: PropTypes.string.isRequired,
    /** Adds props for buttons eg.: [example of how to use]. Check buttons page for more information. */
    buttonProps: PropTypes.shape({
      /** Alert's text button label. */
      text: PropTypes.string,
      /** Action to be executed when the button is clicked. */
      onClick: PropTypes.func,
    }),
  };

  static defaultProps = {
    type: "info",
    buttonProps: { text: "", onClick: () => {} },
  };

  icon = () => {
    const { icon } = this.props;
    return icon ? (
      <Icon type={icon} color="mineral-70" className="lab-alert__icon" />
    ) : undefined;
  };

  button = () => {
    const {
      buttonProps: { text },
    } = this.props;
    return text ? (
      <TextButton size="normal" skin="dark" text={text} />
    ) : undefined;
  };

  handleClick = (event) => {
    const { buttonProps } = this.props;
    if (!isUndefined(buttonProps.onClick)) {
      buttonProps.onClick(event);
    }
  };

  render() {
    const { text, type } = this.props;
    return (
      <div className={`lab-alert lab-alert--${type}`}>
        {this.icon()}
        <span className="lab-alert__message">{text}</span>
        <span
          className="lab-alert__button"
          onClick={this.handleClick}
          role="presentation"
        >
          {this.button()}
        </span>
      </div>
    );
  }
}
