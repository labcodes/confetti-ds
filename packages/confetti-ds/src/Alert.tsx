import React from "react";
import { isUndefined } from "lodash";

import Icon from "./Icon";
import TextButton, { TextButtonProps } from "./Button/TextButton";

import { IconTypes } from "./constants";
export interface AlertProps {
  /** This is the message text string. */
  text: string;
  /** Type of the Alert. */
  type?: "info" | "warn" | "error";
  /** Sets the icon related to the alertâ€™s message. */
  icon?: IconTypes;
  /** Adds props for buttons e.g.: [example of how to use]. Check buttons page for more information. */
  buttonProps?: TextButtonProps;
}

export default function Alert({
  text,
  type = "info",
  icon,
  buttonProps = { text: "", onClick: () => {} },
}: AlertProps) {
  const handleClick = (event) => {
    if (!isUndefined(buttonProps.onClick)) {
      buttonProps.onClick(event);
    }
  };

  return (
    <div className={`lab-alert lab-alert--${type}`}>
      {icon ? <Icon type={icon} className="lab-alert__icon" /> : undefined}
      <span className="lab-alert__message">{text}</span>
      <span
        className="lab-alert__button"
        onClick={handleClick}
        role="presentation"
      >
        {text ? (
          <TextButton size="normal" skin="dark" text={text} />
        ) : undefined}
      </span>
    </div>
  );
}
