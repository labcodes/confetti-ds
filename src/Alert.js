import React from "react";
import PropTypes from "prop-types";
import { isUndefined } from "lodash";

import Icon from "./Icon";
import TextButton from "./Button/TextButton";
import { ICON_TYPES } from "./constants";

export default function Alert({ text, type, icon, buttonProps }) {
  const handleIcon = () =>
    icon ? (
      <Icon type={icon} color="mineral-70" className="lab-alert__icon" />
    ) : undefined;

  const button = () =>
    text ? <TextButton size="normal" skin="dark" text={text} /> : undefined;

  const handleClick = (event) => {
    if (!isUndefined(buttonProps.onClick)) {
      buttonProps.onClick(event);
    }
  };

  return (
    <div className={`lab-alert lab-alert--${type}`}>
      {handleIcon()}
      <span className="lab-alert__message">{text}</span>
      <span
        className="lab-alert__button"
        onClick={handleClick}
        role="presentation"
      >
        {button()}
      </span>
    </div>
  );
}

Alert.propTypes = {
  /** This is the message text string. */
  text: PropTypes.string.isRequired,
  /** Type of the Alert. */
  type: PropTypes.oneOf(["info", "warn", "error"]),
  /** Sets the icon related to the alertâ€™s message. */
  icon: PropTypes.oneOf(ICON_TYPES).isRequired,
  /** Adds props for buttons e.g.: [example of how to use]. Check buttons page for more information. */
  buttonProps: PropTypes.shape({
    /** Alert's text button label. */
    text: PropTypes.string,
    /** Action to be executed when the button is clicked. */
    onClick: PropTypes.func,
  }),
};

Alert.defaultProps = {
  type: "info",
  buttonProps: { text: "", onClick: () => {} },
};
