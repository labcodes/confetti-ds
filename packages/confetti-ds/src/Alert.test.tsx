import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Alert from "./Alert";

describe("Alert", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<Alert text="Error Alert" type="error" icon="magnifying-glass" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with button", () => {
    const renderedComponent = renderer
      .create(
        <Alert
          text="Info Alert with button"
          type="warn"
          icon="wallet"
          buttonProps={{
            label: "Alert Button",
            onClick: () => {},
          }}
        />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("calls props.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const mountedAlert = mount(
      <Alert
        text="Test click on Alert"
        type="info"
        icon="calendar"
        buttonProps={{
          label: "Info Button",
          onClick: mockOnClick,
        }}
      />
    );
    expect(mockOnClick.mock.calls.length).toEqual(0);
    mountedAlert.find("button").simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
