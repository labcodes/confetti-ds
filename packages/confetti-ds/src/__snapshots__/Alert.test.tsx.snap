// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders with base props 1`] = `
<div
  className="lab-alert lab-alert--error"
>
  <span
    className="lab-icon lab-icon--Search  lab-alert__icon"
  >
    <svg
      aria-hidden={true}
      className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
      fill="currentColor"
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.5 3a5.5 5.5 0 0 1 4.23 9.02l4.12 4.13a.5.5 0 0 1-.63.76l-.07-.06-4.13-4.12A5.5 5.5 0 1 1 8.5 3Zm0 1a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    className="lab-alert__message"
  >
    Error Alert
  </span>
  <span
    className="lab-alert__button"
    onClick={[Function]}
    role="presentation"
  >
    <button
      className="lab-btn lab-btn--text lab-btn--normal lab-btn--dark"
      onClick={[Function]}
      type="button"
    >
      
      Error Alert
    </button>
  </span>
</div>
`;

exports[`Alert renders with button 1`] = `
<div
  className="lab-alert lab-alert--warn"
>
  <span
    className="lab-icon lab-icon--Wallet  lab-alert__icon"
  >
    <svg
      aria-hidden={true}
      className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
      fill="currentColor"
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.5 11a.5.5 0 0 0 0 1h1a.5.5 0 0 0 0-1h-1ZM3 4.5C3 3.67 3.67 3 4.5 3H14a2 2 0 0 1 2 2v.27c.6.34 1 .99 1 1.73v8a2 2 0 0 1-2 2H5.5A2.5 2.5 0 0 1 3 14.5v-10ZM14 4H4.5a.5.5 0 0 0 0 1H15a1 1 0 0 0-1-1ZM4.5 6a1.5 1.5 0 0 1-.5-.09v8.59c0 .83.67 1.5 1.5 1.5H15a1 1 0 0 0 1-1V7a1 1 0 0 0-1-1H4.5Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    className="lab-alert__message"
  >
    Info Alert with button
  </span>
  <span
    className="lab-alert__button"
    onClick={[Function]}
    role="presentation"
  >
    <button
      className="lab-btn lab-btn--text lab-btn--normal lab-btn--dark"
      onClick={[Function]}
      type="button"
    >
      
      Info Alert with button
    </button>
  </span>
</div>
`;
